build-bench-env.sh

```diff
@@ -192,6 +198,25 @@ function write_version {  # name, git-tag, repo
   echo "$1: $2, $commit, $3" > "$devdir/version_$1.txt"
 }
 
+function partial_checkout {  # name, git-tag, directory, git repo, directory to download
+  phase "build $1: version $2"
+  pushd $devdir
+  if test "$rebuild" = "1"; then
+    rm -rf "$3"
+  fi
+  if test -d "$3"; then
+    echo "$devdir/$3 already exists; no need to git clone"
+    cd "$3"
+  else
+    git clone --depth=1 --filter=blob:none --no-checkout $4 $3
```

Comment:
This pulls enough information to construct each branch on the LLVM repo, which is still huge.  Following your example, I tried to do this locally, and found the following sequence got a pretty fast checkout:
```sh
mkdir scudo
cd scudo
git init
git remote add origin https://github.com/llvm/llvm-project
git fetch origin main --depth 1    #  Only fetches the tip of main.
git sparse-checkout set compiler-rt/lib/scudo/standalone
git checkout main
```
This should also work, if we switch to a particular commit id for Scudo, which would provide more reproducible benchmarking.

Reply:
```
git remote add origin https://github.com/llvm/llvm-project
git fetch origin main --depth 1    #  Only fetches the tip of main.
```
^ This will download every single file present on the tip of the `main` branch, instead of only the ones in `compiler-rt/lib/scudo/standalone`, which is way bigger than only the branches metadata.

Reply:
Okay, so it did it much quicker on my Azure machine, than your script which filled VMs disk.  But I think the following sequence might be better.
```sh
mkdir scudo
cd scudo
git init
git remote add origin https://github.com/llvm/llvm-project
git config extensions.partialClone origin
git fetch origin main --depth 1 --filter=blob:none   #  Only fetches the tip of main.
git sparse-checkout set compiler-rt/lib/scudo/standalone
git checkout main
```
which resulted in a checkout in a few seconds with
```sh
scudo$ du -m | grep -v "/.*/"
18      ./.git
1       ./compiler-rt
18      .
```